File: backend-java/src/main/java/com/java_services/backend_java/account/domain/repositories/passwordResetToken/db/PasswordResetTokenRepository.java
package com.java_services.backend_java.account.domain.repositories.passwordResetToken.db;

import org.springframework.stereotype.Repository;
import com.java_services.backend_java.account.domain.entities.passwordResetToken.PasswordResetToken;
import com.java_services.backend_java.account.domain.interfaces.Database;

@Repository
public class PasswordResetTokenRepository {

    private final CreatePasswordResetTokenRepository createRepo;
    private final DeletePasswordResetTokenRepository deleteRepo;
    private final ReadPasswordResetTokenRepository readRepo;
    public PasswordResetTokenRepository(Database database) {
        this.createRepo = new CreatePasswordResetTokenRepository(database);
        this.deleteRepo = new DeletePasswordResetTokenRepository(database);
        this.readRepo = new ReadPasswordResetTokenRepository(database);
    }

    public Long create(PasswordResetToken token) {
        return createRepo.create(token);
    }

    public PasswordResetToken getByToken(String token) {
        return readRepo.getByToken(token); 
    }
    
    public int delete(PasswordResetToken token) {
        return deleteRepo.delete(token.getToken());
    }  
}

File: backend-javascript/src/account/domain/repositories/password-reset-token/database/password.reset.token.repository.ts
import { Inject, Injectable } from '@nestjs/common';
import { CreateRepository } from './create/create.repository';
import { PasswordResetToken } from '@src/account/domain/entities/password-reset-token/password.reset.token.entity';
import type { DatabaseAdapterInterface } from '@src/account/domain/interfaces/database.adapter.interface';
import { UpdatePasswordResetTokenRepository } from './update/update.password.reset.token.repository';
import { GetByHashTokenRepository } from './get/get.by.hash.token.repository';


@Injectable()
export class PasswordResetTokenRepository {

  private createRepository: CreateRepository
  private updateRepository: UpdatePasswordResetTokenRepository
  private getByHashTokenRepository: GetByHashTokenRepository

  constructor(@Inject('DatabaseAdapterInterface') private readonly db: DatabaseAdapterInterface) {
    this.createRepository = new CreateRepository(this.db)
    this.updateRepository = new UpdatePasswordResetTokenRepository(this.db)
    this.getByHashTokenRepository = new GetByHashTokenRepository(this.db)
  }

  async create(passwordResetToken: PasswordResetToken): Promise<number> {
    return this.createRepository.handle(passwordResetToken);
  }

  async getByHashToken(hashToken: string): Promise<PasswordResetToken | null> {
    return this.getByHashTokenRepository.handle(hashToken);
  }

  async update(passwordResetToken: PasswordResetToken): Promise<number>{
    return this.updateRepository.handle(passwordResetToken)
  }
}